docs/internals/contributing/committing-code.txt

```diff
@@ -29,58 +29,59 @@ to standard themselves.
 
 .. _Jenkins wiki page: https://code.djangoproject.com/wiki/Jenkins
 
-Here is one way to commit a pull request::
+An easy way to checkout a pull request locally is to use the following script
+(``upstream`` is assumed to be ``django/django``):
 
-    # Create a new branch tracking upstream/master -- upstream is assumed
-    # to be django/django.
-    git checkout -b pull_xxxxx upstream/master
+.. code-block:: bash
 
-    # Download the patches from github and apply them.
-    curl https://github.com/django/django/pull/xxxxx.patch | git am
+    #!/bin/sh
+    git fetch upstream pull/$1/head:pr/$1
+    git checkout pr/$1
+
+Put this script on your ``PATH`` with a name like ``ghpr`` and then you can
+simply run ``ghpr ####`` to checkout the corresponding pull request.
 
 At this point, you can work on the code. Use ``git rebase -i`` and ``git
 commit --amend`` to make sure the commits have the expected level of quality.
-Once you're ready::
-
-    # Make sure master is ready to receive changes.
-    git checkout master
-    git pull upstream master
-    # Merge the work as "fast-forward" to master, to avoid a merge commit.
-    git merge --ff-only pull_xxxxx
-    # Check that only the changes you expect will be pushed to upstream.
-    git push --dry-run upstream master
-    # Push!
-    git push upstream master
-
-    # Get rid of the pull_xxxxx branch.
-    git branch -d pull_xxxxx
-
-An alternative is to add the contributor's repository as a new remote,
-checkout the branch and work from there::
-
-    git remote add <contributor> https://github.com/<contributor>/django.git
-    git checkout pull_xxxxx <contributor> <contributor's pull request branch>
-
-Yet another alternative is to fetch the branch without adding the
-contributor's repository as a remote::
-
-    git fetch https://github.com/<contributor>/django.git <contributor's pull request branch>
-    git checkout -b pull_xxxxx FETCH_HEAD
-
-At this point, you can work on the code and continue as above.
-
-GitHub provides a one-click merge functionality for pull requests. This should
-only be used if the pull request is 100% ready, and you have checked it for
-errors (or trust the request maker enough to skip checks). Currently, it isn't
-possible to check that the tests pass and that the docs build without
-downloading the changes to your development environment.
+Once you're ready:
+
+.. code-block:: console
+
+    $ # Pull in the latest changes from master.
+    $ git checkout master
+    $ git pull upstream master
+    $ # Rebase the pull request on master.
+    $ git checkout pr/####
+    $ git rebase master
+    $ # Merge the work as "fast-forward" to master to avoid a merge commit.
+    $ # (in practice, you can omit "--ff-only" since you just rebased)
+    $ git merge --ff-only pr/XXXX
```

Comment:
Yes, cherry-pick is fine, but I use rebase/merge to avoid going to the mouse to copy/paste the hash (although I imagine there's a way to avoid that). Obviously, this is just a sample workflow and committers are welcome to do whatever they like.